src/privatebin.rs

```diff
@@ -74,33 +74,34 @@ pub struct DecryptedPaste {
     pub attachment_name: Option<String>,
 }
 
-#[derive(Deserialize, Debug, Serialize)]
+#[derive(Deserialize, Debug, Serialize, Clone)]
 pub struct PostPasteResponse {
     pub deletetoken: String,
     pub id: String,
     pub status: u32,
     pub url: String,
+    pub baseurl: String,
```

Comment:
Would it make sense to change this to be of type URL?

This would:

1. Us not needing to convert it to a string (Since it is already a URL)
2. And in turn, we would not need to convert it back to a URL after.

Currently we convert it to a string and back to an URL later.

Reply:
Hi, yes, I think that would be better and simpler. 

I was consciously leaning towards using std / locally defined types (as opposed to external) in the structs as it helps make uniffi export easier. 
But a URL is simple enough to translate, and am already doing that in my [uniffi branch](https://github.com/nain-F49FF806/pbcli/tree/uniffi). 

So there doesn't seem to be any benefit in avoiding it. 
I'll send a patch your way using the url type.

Reply:
On second thought (and from attempting to change it), the url type does not have Serialize/Deserialize trait. 
So if it were included, then the derive on `PostPasteResponse` won't work. And we'll have to do it manually.

The implementation for this would require us to do the String -> Url, Url -> String anway. 
So, I think this might just be simpler overall. What do you think?

Reply:
Greetings,

hmm. I get your point and I would not have any pain merging this as is. The good part about converting it to a URL would be:

1. The serialize/deserialize takes place in one location
2. It also prevents developers from moving illegal (non-url strings) into the struct

Tell me what you think and I can proceed to merge it

Reply:
Greetings, 

The advantages you listed do sound good. Unfortunately, I am not well versed in writing custom deserializers yet. So am unable to provide a working implementation for the same. If you will, feel free to do so.

In this case, either choice (url/string) will not have any effect on uniffi bindings, so no worries there. :)

Regards.

Reply:
p.s: 

In my attempt, I was able to get it to serialize, like so. 

```rust
#[derive(Deserialize, Debug, Serialize, Clone)]
pub struct PostPasteResponse {
    pub deletetoken: String,
    pub id: String,
    pub status: u32,
    pub url: String,
    #[serde(serialize_with = "PostPasteResponse::serialize_baseurl")]
    pub baseurl: Url,
    pub bs58key: String,
}

impl PostPasteResponse {
    pub fn serialize_baseurl<S>(url: &Url, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        serializer.serialize_str(url.as_str())
    }
}
```

But couldn't figure how to get it working with deserialization.

Reply:
Greetings,

I allowed myself to push the (de)serialization into the PR branch. If you could check that it works for you on your end I would proceed to merge the PR.

Best Regards
Mydayyy

Reply:
Greetings. 

I have tested these changes and can confirm it works. :+1:  
The solution you provided is much simpler than I thought it would be.  :smile: :upside_down_face: 
Please go ahead with the merge. 

Regards.